server/modules/module_def.js

```diff
@@ -129,75 +130,109 @@ let loadModuleAtPath = function(path) {
   });
 };
 
-class ModuleDefinitionLibrary {
-  constructor() {
-    // map of path -> promise<string>
-    // This map is only populated during the loading process.
-    this.moduleLoader = {};
-
-    // map of path -> string.
-    this.modules = {};
+/**
+ * The ModuleDef class contains all the information necessary to load & 
+ * instantiate a module, including code location and config parameters.
+ */
+class ModuleDef {
```

Comment:
You should make sure that this and the client equivalent have the same name. I can't remember what I called the class when I refactored the client.